src/main/resources/META-INF/rewrite/replace-MediaType-literals.yml

```diff
@@ -0,0 +1,86 @@
+#
+# Copyright 2023 the original author or authors.
+# <p>
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://www.apache.org/licenses/LICENSE-2.0
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.spring.http.ReplaceStringLiteralsWithMediaTypeConstants
+displayName: Replace String literals with `MediaType` constants
+description: Replace String literals with `org.springframework.http.MediaType` constants.
+applicability:
+  anySource:
+    - org.openrewrite.maven.search.FindDependency:
+        groupId: org.springframework
+        artifactId: spring-web
```

Comment:
This fails to pick up transitive dependencies; I see two quick options, but both require changes in rewrite:

1. Add an option to FindDependency to allow picking up transitive dependencies
2. Switch to DependencyInsight, and add an option there to generate markers

Any other ideas or preferences?